test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs

```diff
@@ -493,6 +493,208 @@ private static void CreateXmlFile(string configFilePath, string nugetConfigStrin
+
+                // Assert
+                Assert.Equal(shouldSucceed, result.ExitCode == 0);
+
+                // Assert error message
+                if (!shouldSucceed)
+                {
+                    string expectedErrorMessage = "error: The protocol version specified is invalid.";
+                    VerifyResultFailure(result, expectedErrorMessage);
+                }
+            }
+        }
+
+        [PlatformFact(Platform.Windows)]
+        public void Sources_WhenUpdatingSourceWithProtocolVersion_GotUpdated()
+        {
+            using (TestDirectory configFileDirectory = _fixture.CreateTestDirectory())
+            {
+                string configFileName = "nuget.config";
+                string configFilePath = Path.Combine(configFileDirectory, configFileName);
+
+                var nugetConfig =
+                    @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <packageSources>
+    <add key=""test_source"" value=""https://source.test.initial"" />
+  </packageSources>
+</configuration>";
+                CreateXmlFile(configFilePath, nugetConfig);
+
+                ISettings settings = Settings.LoadDefaultSettings(
+                    configFileDirectory,
+                    configFileName,
+                    null);
+
+                PackageSourceProvider packageSourceProvider = new PackageSourceProvider(settings);
+                var sources = packageSourceProvider.LoadPackageSources().ToList();
+                Assert.Single(sources);
+
+                PackageSource source = sources.Single();
+                Assert.Equal("test_source", source.Name);
+                Assert.Equal("https://source.test.initial", source.Source);
+
+                // Arrange
+                var args = new string[]
+                {
+                    "nuget",
+                    "update",
+                    "source",
+                    "test_source",
+                    "--source",
+                    "https://source.test",
+                    "--configfile",
+                    configFilePath,
+                    "--protocol-version",
+                    "3"
+                };
+
+                // Act
+                CommandRunnerResult result = _fixture.RunDotnet(configFileDirectory, string.Join(" ", args), ignoreExitCode: true);
+
+                // Assert
+                Assert.True(result.Success, result.Output + " " + result.Errors);
+
+                ISettings loadedSettings = Settings.LoadDefaultSettings(root: configFileDirectory, configFileName: null, machineWideSettings: null);
+
+                SettingSection packageSourcesSection = loadedSettings.GetSection("packageSources");
+                SourceItem sourceItem = packageSourcesSection?.GetFirstItemWithAttribute<SourceItem>("key", "test_source");
+                Assert.Equal("3", sourceItem.ProtocolVersion);
+            }
+        }
+
+        [PlatformTheory(Platform.Windows)]
```

Comment:
By the way why we're only testing on Windows?

Reply:
I have used the same attribute as other tests in this file for consistency. The only platform that those are marked for is Windows.
Shall I change whole class or only newly added tests?

Reply:
In that case, please keep it as it's now.